### PR TITLE
E4.5 - Simple patient documentation search

### PR DESCRIPTION
**Pull request by nikolagudelj**

### CODE REVIEW

**Comment by nikolagudelj** [comment]
[[E4.3] As a patient I want to be able to search through medical record, so that I choose data that I want to observe. [3]](https://trello.com/c/Gtfp3oAT/25-e43-as-a-patient-i-want-to-be-able-to-search-through-medical-record-so-that-i-choose-data-that-i-want-to-observe-3)

File: HealthcareBase/Repository/UsersRepository/EmployeesAndPatientsRepository/Interface/MedicalRecordRepository.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿using Model.Users.Patient.MedicalHistory;
+using Repository.Generics;
+
+namespace Repository.UsersRepository.EmployeesAndPatientsRepository
+{
+    public interface MedicalRecordRepository
+    {
+
```
Comment (rammba) [inline HealthcareBase/Repository/UsersRepository/EmployeesAndPatientsRepository/Interface/MedicalRecordRepository.cs:8]: Rename all interfaces to start with "I" like (IMedicalRecordRepo). This repeats in few repos.

File: HealthcareBase/Service/ScheduleService/ProcedureService/DoctorCredentialsDto.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        public string Name { get; set; }
+        public string Surname { get; set; }
+
+        protected override void ConfigureFilter()
+        {
+            if (!string.IsNullOrEmpty(Name)) 
+                AddExpressionFunction(examination => examination.Doctor.Person.Name.Contains(Name));
+            if (!string.IsNullOrEmpty(Surname))
+                AddExpressionFunction(examination => examination.Doctor.Person.Surname.Contains(Surname));
+        }
```
Comment (hi-imAndy) [inline HealthcareBase/Service/ScheduleService/ProcedureService/DoctorCredentialsDto.cs:21]: Violated Law of Demeter - "use only one dot"

**Review by hi-imAndy — CHANGES_REQUESTED** [review]
Check comments

**Comment by rammba** [comment]
[(1) PR Simple patient documentation search](https://trello.com/c/Va5zyVmP/81-1-pr-simple-patient-documentation-search)

File: HealthcareBase/Repository/UsersRepository/EmployeesAndPatientsRepository/Interface/MedicalRecordRepository.cs (hunk 1 above)
Comment (sStrbacki) [reply HealthcareBase/Repository/UsersRepository/EmployeesAndPatientsRepository/Interface/MedicalRecordRepository.cs]: All interfaces renamed.

File: HealthcareBase/Service/ScheduleService/ProcedureService/DoctorCredentialsDto.cs (hunk 2 above)
Comment (nikolagudelj) [reply HealthcareBase/Service/ScheduleService/ProcedureService/DoctorCredentialsDto.cs:17]: Advanced query requires access to doctor's surname, therefore it must be accessed this way.

**Review by rammba — DISMISSED** [review]

**Review by rammba — APPROVED** [review]

**Review by hi-imAndy — APPROVED** [review]

**Pull request merged**